src/GraphQL/Introspection/ISchemaComparer.cs

```diff
@@ -0,0 +1,129 @@
+using System.Collections.Generic;
+using GraphQL.Types;
+
+namespace GraphQL.Introspection
+{
+    /// <summary>
+    /// Provides the ability to order the schema elements upon introspection.
+    /// </summary>
+    public interface ISchemaComparer
+    {
+        /// <summary>
+        /// Returns a comparer for GraphQL types.
+        /// </summary>
+        IComparer<IGraphType> TypeComparer { get; }
+
+        /// <summary>
+        /// Returns a comparer for fields withing enclosing type.
+        /// </summary>
+        /// <param name="parent"> Parent type to which the fields belong. </param>
+        IComparer<IFieldType> FieldComparer(IGraphType parent);
+
+        /// <summary>
+        /// Returns a comparer for field arguments.
+        /// </summary>
+        /// <param name="field"> The field to which the arguments belong. </param>
+        IComparer<QueryArgument> ArgumentComparer(IFieldType field);
+
+        /// <summary>
+        /// Returns a comparer for enum values.
+        /// </summary>
+        /// <param name="parent"> The enumeration to which the enum values belong. </param>
+        IComparer<EnumValueDefinition> EnumValueComparer(EnumerationGraphType parent);
+
+        /// <summary>
+        /// Returns a comparer for GraphQL directives.
+        /// </summary>
+        IComparer<DirectiveGraphType> DirectiveComparer { get; }
+    }
+
+    /// <summary>
+    /// Default schema comparer. By default all elements are returned as is, no sorting is applied.
+    /// </summary>
+    public class DefaultSchemaComparer : ISchemaComparer
```

Comment:
@Shane32 we have already discussed this. I am in favor of removing the default sorting in v4. `LegacyV3SchemaComparer` can be used to switch to the old behavior.

Reply:
Sounds fine to me.